packages/vue-app/template/observer.js

```diff
@@ -0,0 +1,49 @@
+// Inspired by: https://github.com/nuxt/image/blob/dev/src/runtime/utils/observer.ts
+
+let observerInstance
+let observerIdCtr = 1
+const OBSERVER_ID_KEY = '__nuxt_prefetch_observer__'
+
+export function get () {
+  if (typeof IntersectionObserver === 'undefined') {
+    return false
+  }
+
+  if (observerInstance) {
+    return observerInstance
+  }
+
+  const onIntersect = (entries) => {
+    entries.forEach(({ intersectionRatio, target: link }) => {
+      if (intersectionRatio <= 0 || !link.__prefetch) {
```

Comment:
Target is not necessarily link if util is refactored

Reply:
WDYT - should we keep the observer more general or specifically for prefetching only?

Reply:
We are exposing it to global context namespace so i think makes sense (to ensure forward api compatibility) :D